<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Accumulation of deleterious instructions | Supplemental Material for Adaptive phenotypic plasticity stabilizes evolution in fluctuating environments</title>
  <meta name="description" content="Supplemental material" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Accumulation of deleterious instructions | Supplemental Material for Adaptive phenotypic plasticity stabilizes evolution in fluctuating environments" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Supplemental material" />
  <meta name="github-repo" content="amlalejini/evolutionary-consequences-of-plasticity" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Accumulation of deleterious instructions | Supplemental Material for Adaptive phenotypic plasticity stabilizes evolution in fluctuating environments" />
  
  <meta name="twitter:description" content="Supplemental material" />
  

<meta name="author" content="Alexander Lalejini, Austin J. Ferguson, Nkrumah A. Grant, and Charles Ofria" />


<meta name="date" content="2021-05-22" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="evolution-and-maintenance-of-novel-traits.html"/>
<link rel="next" href="regulation-in-avida.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Supplemental material</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#about-our-supplemental-material"><i class="fa fa-check"></i><b>1.1</b> About our supplemental material</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#contributing-authors"><i class="fa fa-check"></i><b>1.2</b> Contributing authors</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#research-overview"><i class="fa fa-check"></i><b>1.3</b> Research overview</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-availability.html"><a href="data-availability.html"><i class="fa fa-check"></i><b>2</b> Data availability</a><ul>
<li class="chapter" data-level="2.1" data-path="data-availability.html"><a href="data-availability.html#source-code"><i class="fa fa-check"></i><b>2.1</b> Source code</a></li>
<li class="chapter" data-level="2.2" data-path="data-availability.html"><a href="data-availability.html#experimental-results"><i class="fa fa-check"></i><b>2.2</b> Experimental results</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="compile-and-run-experiments-locally.html"><a href="compile-and-run-experiments-locally.html"><i class="fa fa-check"></i><b>3</b> Compile and run experiments locally</a><ul>
<li class="chapter" data-level="3.1" data-path="compile-and-run-experiments-locally.html"><a href="compile-and-run-experiments-locally.html#docker"><i class="fa fa-check"></i><b>3.1</b> Docker</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="validation-experiment.html"><a href="validation-experiment.html"><i class="fa fa-check"></i><b>4</b> Validation experiment</a><ul>
<li class="chapter" data-level="4.1" data-path="validation-experiment.html"><a href="validation-experiment.html#overview"><i class="fa fa-check"></i><b>4.1</b> Overview</a></li>
<li class="chapter" data-level="4.2" data-path="validation-experiment.html"><a href="validation-experiment.html#analysis-dependencies"><i class="fa fa-check"></i><b>4.2</b> Analysis dependencies</a></li>
<li class="chapter" data-level="4.3" data-path="validation-experiment.html"><a href="validation-experiment.html#setup"><i class="fa fa-check"></i><b>4.3</b> Setup</a></li>
<li class="chapter" data-level="4.4" data-path="validation-experiment.html"><a href="validation-experiment.html#evolution-of-phenotypic-plasticity"><i class="fa fa-check"></i><b>4.4</b> Evolution of phenotypic plasticity</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="evolutionary-change.html"><a href="evolutionary-change.html"><i class="fa fa-check"></i><b>5</b> Evolutionary change</a><ul>
<li class="chapter" data-level="5.1" data-path="evolutionary-change.html"><a href="evolutionary-change.html#overview-1"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="evolutionary-change.html"><a href="evolutionary-change.html#analysis-dependencies-1"><i class="fa fa-check"></i><b>5.2</b> Analysis dependencies</a></li>
<li class="chapter" data-level="5.3" data-path="evolutionary-change.html"><a href="evolutionary-change.html#setup-1"><i class="fa fa-check"></i><b>5.3</b> Setup</a></li>
<li class="chapter" data-level="5.4" data-path="evolutionary-change.html"><a href="evolutionary-change.html#the-evolution-of-phenotypic-plasticity"><i class="fa fa-check"></i><b>5.4</b> The evolution of phenotypic plasticity</a></li>
<li class="chapter" data-level="5.5" data-path="evolutionary-change.html"><a href="evolutionary-change.html#average-generation"><i class="fa fa-check"></i><b>5.5</b> Average generation</a></li>
<li class="chapter" data-level="5.6" data-path="evolutionary-change.html"><a href="evolutionary-change.html#coalescence-event-count"><i class="fa fa-check"></i><b>5.6</b> Coalescence event count</a><ul>
<li class="chapter" data-level="5.6.1" data-path="evolutionary-change.html"><a href="evolutionary-change.html#average-number-of-generations-between-coalescence-events"><i class="fa fa-check"></i><b>5.6.1</b> Average number of generations between coalescence events</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="evolutionary-change.html"><a href="evolutionary-change.html#phenotypic-volatility-along-the-dominant-lineage"><i class="fa fa-check"></i><b>5.7</b> Phenotypic volatility along the dominant lineage</a><ul>
<li class="chapter" data-level="5.7.1" data-path="evolutionary-change.html"><a href="evolutionary-change.html#phenotypic-volatility-normalized-by-generations-elapsed"><i class="fa fa-check"></i><b>5.7.1</b> Phenotypic volatility normalized by generations elapsed</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="evolutionary-change.html"><a href="evolutionary-change.html#phenotypic-fidelity"><i class="fa fa-check"></i><b>5.8</b> Phenotypic fidelity</a></li>
<li class="chapter" data-level="5.9" data-path="evolutionary-change.html"><a href="evolutionary-change.html#mutation-count"><i class="fa fa-check"></i><b>5.9</b> Mutation count</a><ul>
<li class="chapter" data-level="5.9.1" data-path="evolutionary-change.html"><a href="evolutionary-change.html#mutation-count-normalized-by-generations-elapsed"><i class="fa fa-check"></i><b>5.9.1</b> Mutation count normalized by generations elapsed</a></li>
</ul></li>
<li class="chapter" data-level="5.10" data-path="evolutionary-change.html"><a href="evolutionary-change.html#genotypic-fidelity"><i class="fa fa-check"></i><b>5.10</b> Genotypic fidelity</a></li>
<li class="chapter" data-level="5.11" data-path="evolutionary-change.html"><a href="evolutionary-change.html#characterizing-variation-along-dominant-lineages"><i class="fa fa-check"></i><b>5.11</b> Characterizing variation along dominant lineages</a><ul>
<li class="chapter" data-level="5.11.1" data-path="evolutionary-change.html"><a href="evolutionary-change.html#mutational-instability"><i class="fa fa-check"></i><b>5.11.1</b> Mutational instability</a></li>
<li class="chapter" data-level="5.11.2" data-path="evolutionary-change.html"><a href="evolutionary-change.html#mutational-stability"><i class="fa fa-check"></i><b>5.11.2</b> Mutational stability</a></li>
<li class="chapter" data-level="5.11.3" data-path="evolutionary-change.html"><a href="evolutionary-change.html#for-plastic-populations-what-fraction-of-phenotype-altering-mutations-occurred-in-the-unexpressed-phenotype"><i class="fa fa-check"></i><b>5.11.3</b> For PLASTIC populations, what fraction of phenotype-altering mutations occurred in the unexpressed phenotype?</a></li>
<li class="chapter" data-level="5.11.4" data-path="evolutionary-change.html"><a href="evolutionary-change.html#for-plastic-populations-what-fraction-of-mutations-that-affect-the-unexpressed-phenotype-are-deleterious-versus-beneficial"><i class="fa fa-check"></i><b>5.11.4</b> For PLASTIC populations, what fraction of mutations that affect the unexpressed phenotype are deleterious versus beneficial?</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="evolutionary-change.html"><a href="evolutionary-change.html#manuscript-figures"><i class="fa fa-check"></i><b>5.12</b> Manuscript figures</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html"><i class="fa fa-check"></i><b>6</b> Evolution and maintenance of novel traits</a><ul>
<li class="chapter" data-level="6.1" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#overview-2"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#analysis-dependencies-2"><i class="fa fa-check"></i><b>6.2</b> Analysis dependencies</a></li>
<li class="chapter" data-level="6.3" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#setup-2"><i class="fa fa-check"></i><b>6.3</b> Setup</a></li>
<li class="chapter" data-level="6.4" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#the-evolution-of-phenotypic-plasticity-1"><i class="fa fa-check"></i><b>6.4</b> The evolution of phenotypic plasticity</a></li>
<li class="chapter" data-level="6.5" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#final-novel-task-count-dominant-genotype"><i class="fa fa-check"></i><b>6.5</b> Final novel task count (dominant genotype)</a></li>
<li class="chapter" data-level="6.6" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#novel-task-count-final-population"><i class="fa fa-check"></i><b>6.6</b> Novel task count (final population)</a></li>
<li class="chapter" data-level="6.7" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#novel-task-discovery-lineage"><i class="fa fa-check"></i><b>6.7</b> Novel task discovery (lineage)</a></li>
<li class="chapter" data-level="6.8" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#novel-task-discovery-population"><i class="fa fa-check"></i><b>6.8</b> Novel task discovery (population)</a></li>
<li class="chapter" data-level="6.9" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#novel-task-discovery-frequency-lineage"><i class="fa fa-check"></i><b>6.9</b> Novel task discovery frequency (lineage)</a></li>
<li class="chapter" data-level="6.10" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#novel-tasks-gained-lineage"><i class="fa fa-check"></i><b>6.10</b> Novel tasks gained (lineage)</a></li>
<li class="chapter" data-level="6.11" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#novel-task-loss-lineage"><i class="fa fa-check"></i><b>6.11</b> Novel task loss (lineage)</a></li>
<li class="chapter" data-level="6.12" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#frequency-of-novel-task-loss-lineage"><i class="fa fa-check"></i><b>6.12</b> Frequency of novel task loss (lineage)</a></li>
<li class="chapter" data-level="6.13" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#how-many-instances-of-novel-trait-loss-co-occurred-with-changes-in-base-phenotype"><i class="fa fa-check"></i><b>6.13</b> How many instances of novel trait loss co-occurred with changes in base phenotype?</a></li>
<li class="chapter" data-level="6.14" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#manuscript-figures-1"><i class="fa fa-check"></i><b>6.14</b> Manuscript figures</a></li>
<li class="chapter" data-level="6.15" data-path="evolution-and-maintenance-of-novel-traits.html"><a href="evolution-and-maintenance-of-novel-traits.html#combined-panel"><i class="fa fa-check"></i><b>6.15</b> Combined panel</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html"><i class="fa fa-check"></i><b>7</b> Accumulation of deleterious instructions</a><ul>
<li class="chapter" data-level="7.1" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#overview-3"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#analysis-dependencies-3"><i class="fa fa-check"></i><b>7.2</b> Analysis dependencies</a></li>
<li class="chapter" data-level="7.3" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#setup-3"><i class="fa fa-check"></i><b>7.3</b> Setup</a></li>
<li class="chapter" data-level="7.4" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#evolution-of-phenotypic-plasticity-1"><i class="fa fa-check"></i><b>7.4</b> Evolution of phenotypic plasticity</a></li>
<li class="chapter" data-level="7.5" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#poison-instruction-execution"><i class="fa fa-check"></i><b>7.5</b> Poison instruction execution</a><ul>
<li class="chapter" data-level="7.5.1" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#number-of-replicates-where-final-dominant-genotype-executes-the-poison-instruction"><i class="fa fa-check"></i><b>7.5.1</b> Number of replicates where final dominant genotype executes the poison instruction</a></li>
<li class="chapter" data-level="7.5.2" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#poison-instruction-execution-final-population"><i class="fa fa-check"></i><b>7.5.2</b> Poison instruction execution (final population)</a></li>
<li class="chapter" data-level="7.5.3" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#cummulative-poison-instruction-execution-along-final-dominant-lineages"><i class="fa fa-check"></i><b>7.5.3</b> Cummulative poison instruction execution along final dominant lineages</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#characterizing-mutations-that-increase-poison-instruction-execution"><i class="fa fa-check"></i><b>7.6</b> Characterizing mutations that increase poison instruction execution</a><ul>
<li class="chapter" data-level="7.6.1" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#number-of-offspring-along-dominant-lineage-with-increase-in-poison-instruction-execution"><i class="fa fa-check"></i><b>7.6.1</b> Number of offspring along dominant lineage with increase in poison instruction execution</a></li>
<li class="chapter" data-level="7.6.2" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#frequency-of-increases-in-poison-instruction-execution-lineage"><i class="fa fa-check"></i><b>7.6.2</b> Frequency of increases in poison instruction execution (lineage)</a></li>
<li class="chapter" data-level="7.6.3" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#what-fraction-of-mutations-that-increase-poison-instruction-execution-co-occur-with-base-trait-changes"><i class="fa fa-check"></i><b>7.6.3</b> What fraction of mutations that increase poison instruction execution co-occur with base trait changes?</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#what-fraction-of-poison-execution-increases-occur-in-unexpressed-phenotype-as-cryptic-variation"><i class="fa fa-check"></i><b>7.7</b> What fraction of poison execution increases occur in unexpressed phenotype (as cryptic variation)?</a></li>
<li class="chapter" data-level="7.8" data-path="accumulation-of-deleterious-instructions.html"><a href="accumulation-of-deleterious-instructions.html#manuscript-figures-2"><i class="fa fa-check"></i><b>7.8</b> Manuscript figures</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="regulation-in-avida.html"><a href="regulation-in-avida.html"><i class="fa fa-check"></i><b>8</b> Regulation in Avida</a><ul>
<li class="chapter" data-level="8.1" data-path="regulation-in-avida.html"><a href="regulation-in-avida.html#overview-4"><i class="fa fa-check"></i><b>8.1</b> Overview</a></li>
<li class="chapter" data-level="8.2" data-path="regulation-in-avida.html"><a href="regulation-in-avida.html#analysis-dependencies-4"><i class="fa fa-check"></i><b>8.2</b> Analysis dependencies</a></li>
<li class="chapter" data-level="8.3" data-path="regulation-in-avida.html"><a href="regulation-in-avida.html#setup-4"><i class="fa fa-check"></i><b>8.3</b> Setup</a></li>
<li class="chapter" data-level="8.4" data-path="regulation-in-avida.html"><a href="regulation-in-avida.html#how-many-instructions-do-plastic-genomes-toggle-depending-on-environmental-context"><i class="fa fa-check"></i><b>8.4</b> How many instructions do plastic genomes toggle depending on environmental context?</a></li>
<li class="chapter" data-level="8.5" data-path="regulation-in-avida.html"><a href="regulation-in-avida.html#what-is-the-distrubution-of-toggled-sequence-sizes"><i class="fa fa-check"></i><b>8.5</b> What is the distrubution of toggled sequence sizes?</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html"><i class="fa fa-check"></i><b>9</b> Evolutionary change (variable length genomes)</a><ul>
<li class="chapter" data-level="9.1" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#overview-5"><i class="fa fa-check"></i><b>9.1</b> Overview</a></li>
<li class="chapter" data-level="9.2" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#analysis-dependencies-5"><i class="fa fa-check"></i><b>9.2</b> Analysis dependencies</a></li>
<li class="chapter" data-level="9.3" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#setup-5"><i class="fa fa-check"></i><b>9.3</b> Setup</a></li>
<li class="chapter" data-level="9.4" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#evolution-of-phenotypic-plasticity-2"><i class="fa fa-check"></i><b>9.4</b> Evolution of phenotypic plasticity</a></li>
<li class="chapter" data-level="9.5" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#genome-length"><i class="fa fa-check"></i><b>9.5</b> Genome length</a></li>
<li class="chapter" data-level="9.6" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#average-generation-1"><i class="fa fa-check"></i><b>9.6</b> Average generation</a></li>
<li class="chapter" data-level="9.7" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#coalescence-events"><i class="fa fa-check"></i><b>9.7</b> Coalescence events</a><ul>
<li class="chapter" data-level="9.7.1" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#average-number-of-generations-between-selective-sweeps"><i class="fa fa-check"></i><b>9.7.1</b> Average number of generations between selective sweeps</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#phenotypic-volatility-along-the-dominant-lineage-1"><i class="fa fa-check"></i><b>9.8</b> Phenotypic volatility along the dominant lineage</a></li>
<li class="chapter" data-level="9.9" data-path="evolutionary-change-variable-length-genomes.html"><a href="evolutionary-change-variable-length-genomes.html#mutation-accumulation-along-the-dominant-lineage"><i class="fa fa-check"></i><b>9.9</b> Mutation accumulation along the dominant lineage</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with ‚ù§ and bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Supplemental Material for Adaptive phenotypic plasticity stabilizes evolution in fluctuating environments</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="accumulation-of-deleterious-instructions" class="section level1">
<h1><span class="header-section-number">Chapter 7</span> Accumulation of deleterious instructions</h1>
<p>The effect of adaptive phenotypic plasticity on the accumulation of deleterious genes.</p>
<div id="overview-3" class="section level2">
<h2><span class="header-section-number">7.1</span> Overview</h2>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb315-1" title="1">total_updates &lt;-<span class="st"> </span><span class="dv">200000</span></a>
<a class="sourceLine" id="cb315-2" title="2">replicates &lt;-<span class="st"> </span><span class="dv">100</span></a>
<a class="sourceLine" id="cb315-3" title="3">alpha &lt;-<span class="st"> </span><span class="fl">0.05</span></a>
<a class="sourceLine" id="cb315-4" title="4">focal_poison_penalty &lt;-<span class="st"> </span><span class="fl">0.1</span></a>
<a class="sourceLine" id="cb315-5" title="5"></a>
<a class="sourceLine" id="cb315-6" title="6">focal_traits &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;not&quot;</span>,<span class="st">&quot;nand&quot;</span>,<span class="st">&quot;and&quot;</span>,<span class="st">&quot;ornot&quot;</span>,<span class="st">&quot;or&quot;</span>,<span class="st">&quot;andnot&quot;</span>)</a>
<a class="sourceLine" id="cb315-7" title="7">traits_set_a &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;not&quot;</span>, <span class="st">&quot;and&quot;</span>, <span class="st">&quot;or&quot;</span>)</a>
<a class="sourceLine" id="cb315-8" title="8">traits_set_b &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;nand&quot;</span>, <span class="st">&quot;ornot&quot;</span>, <span class="st">&quot;andnot&quot;</span>)</a>
<a class="sourceLine" id="cb315-9" title="9"></a>
<a class="sourceLine" id="cb315-10" title="10"><span class="co"># Relative location of data.</span></a>
<a class="sourceLine" id="cb315-11" title="11">working_directory &lt;-<span class="st"> &quot;experiments/2021-02-05-hitchhiking/analysis/&quot;</span> <span class="co"># &lt;&lt; For bookdown</span></a>
<a class="sourceLine" id="cb315-12" title="12"><span class="co"># working_directory &lt;- &quot;./&quot;</span></a></code></pre></div>
</div>
<div id="analysis-dependencies-3" class="section level2">
<h2><span class="header-section-number">7.2</span> Analysis dependencies</h2>
<p>Load all required R libraries.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb316-1" title="1"><span class="kw">library</span>(RColorBrewer)</a>
<a class="sourceLine" id="cb316-2" title="2"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb316-3" title="3"><span class="kw">library</span>(rstatix)</a>
<a class="sourceLine" id="cb316-4" title="4"><span class="kw">library</span>(ggsignif)</a>
<a class="sourceLine" id="cb316-5" title="5"><span class="kw">library</span>(scales)</a>
<a class="sourceLine" id="cb316-6" title="6"><span class="kw">library</span>(tidyverse)</a>
<a class="sourceLine" id="cb316-7" title="7"><span class="kw">library</span>(cowplot)</a>
<a class="sourceLine" id="cb316-8" title="8"><span class="kw">library</span>(Hmisc)</a>
<a class="sourceLine" id="cb316-9" title="9"><span class="kw">library</span>(boot)</a>
<a class="sourceLine" id="cb316-10" title="10"><span class="kw">library</span>(fmsb)</a>
<a class="sourceLine" id="cb316-11" title="11"><span class="kw">library</span>(knitr)</a>
<a class="sourceLine" id="cb316-12" title="12"><span class="kw">source</span>(<span class="st">&quot;https://gist.githubusercontent.com/benmarwick/2a1bb0133ff568cbe28d/raw/fb53bd97121f7f9ce947837ef1a4c65a73bffb3f/geom_flat_violin.R&quot;</span>)</a></code></pre></div>
<p>These analyses were conducted/knitted with the following computing environment:</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb317-1" title="1"><span class="kw">print</span>(version)</a></code></pre></div>
<pre><code>##                _                           
## platform       x86_64-pc-linux-gnu         
## arch           x86_64                      
## os             linux-gnu                   
## system         x86_64, linux-gnu           
## status                                     
## major          4                           
## minor          1.0                         
## year           2021                        
## month          05                          
## day            18                          
## svn rev        80317                       
## language       R                           
## version.string R version 4.1.0 (2021-05-18)
## nickname       Camp Pontanezen</code></pre>
</div>
<div id="setup-3" class="section level2">
<h2><span class="header-section-number">7.3</span> Setup</h2>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb319-1" title="1"><span class="co">####### summary data #######</span></a>
<a class="sourceLine" id="cb319-2" title="2">summary_data_loc &lt;-<span class="st"> </span><span class="kw">paste0</span>(working_directory, <span class="st">&quot;data/aggregate.csv&quot;</span>)</a>
<a class="sourceLine" id="cb319-3" title="3">summary_data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(summary_data_loc, <span class="dt">na.strings=</span><span class="st">&quot;NONE&quot;</span>)</a>
<a class="sourceLine" id="cb319-4" title="4"></a>
<a class="sourceLine" id="cb319-5" title="5">summary_data<span class="op">$</span>DISABLE_REACTION_SENSORS &lt;-<span class="st"> </span><span class="kw">as.factor</span>(summary_data<span class="op">$</span>DISABLE_REACTION_SENSORS)</a>
<a class="sourceLine" id="cb319-6" title="6">summary_data<span class="op">$</span>chg_env &lt;-<span class="st"> </span>summary_data<span class="op">$</span>chg_env <span class="op">==</span><span class="st"> &quot;True&quot;</span></a>
<a class="sourceLine" id="cb319-7" title="7">summary_data<span class="op">$</span>dominant_plastic_odd_even &lt;-<span class="st"> </span><span class="kw">as.factor</span>(summary_data<span class="op">$</span>dominant_plastic_odd_even)</a>
<a class="sourceLine" id="cb319-8" title="8">summary_data<span class="op">$</span>sensors &lt;-<span class="st"> </span>summary_data<span class="op">$</span>DISABLE_REACTION_SENSORS <span class="op">==</span><span class="st"> &quot;0&quot;</span></a>
<a class="sourceLine" id="cb319-9" title="9">summary_data<span class="op">$</span>is_plastic &lt;-<span class="st"> </span>summary_data<span class="op">$</span>dominant_plastic_odd_even <span class="op">==</span><span class="st"> &quot;True&quot;</span></a>
<a class="sourceLine" id="cb319-10" title="10">summary_data<span class="op">$</span>POISON_PENALTY &lt;-<span class="st"> </span><span class="kw">as.factor</span>(summary_data<span class="op">$</span>POISON_PENALTY)</a>
<a class="sourceLine" id="cb319-11" title="11"></a>
<a class="sourceLine" id="cb319-12" title="12">summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation &lt;-<span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases <span class="op">/</span><span class="st"> </span>summary_data<span class="op">$</span>dominant_generation_born</a>
<a class="sourceLine" id="cb319-13" title="13">summary_data<span class="op">$</span>frac_hitchhiking_linked_trait_change &lt;-<span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_with_primary_trait_change <span class="op">/</span><span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases</a>
<a class="sourceLine" id="cb319-14" title="14">summary_data<span class="op">$</span>frac_unexpressed_hitchhiker_inc &lt;-<span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_in_unexpressed_phenotype <span class="op">/</span><span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases</a>
<a class="sourceLine" id="cb319-15" title="15">summary_data<span class="op">$</span>frac_expressed_hitchiker_inc &lt;-<span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_in_expressed_phenotype <span class="op">/</span><span class="st"> </span>summary_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases</a>
<a class="sourceLine" id="cb319-16" title="16"></a>
<a class="sourceLine" id="cb319-17" title="17">env_label_fun &lt;-<span class="st"> </span><span class="cf">function</span>(chg_env) {</a>
<a class="sourceLine" id="cb319-18" title="18">  <span class="cf">if</span> (chg_env) {</a>
<a class="sourceLine" id="cb319-19" title="19">    <span class="kw">return</span>(<span class="st">&quot;Fluctuating&quot;</span>)</a>
<a class="sourceLine" id="cb319-20" title="20">  } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb319-21" title="21">    <span class="kw">return</span>(<span class="st">&quot;Constant&quot;</span>)</a>
<a class="sourceLine" id="cb319-22" title="22">  }</a>
<a class="sourceLine" id="cb319-23" title="23">}</a>
<a class="sourceLine" id="cb319-24" title="24"></a>
<a class="sourceLine" id="cb319-25" title="25">sensors_label_fun &lt;-<span class="st"> </span><span class="cf">function</span>(has_sensors) {</a>
<a class="sourceLine" id="cb319-26" title="26">  <span class="cf">if</span> (has_sensors) {</a>
<a class="sourceLine" id="cb319-27" title="27">    <span class="kw">return</span>(<span class="st">&quot;Sensors&quot;</span>)</a>
<a class="sourceLine" id="cb319-28" title="28">  } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb319-29" title="29">    <span class="kw">return</span>(<span class="st">&quot;No sensors&quot;</span>)</a>
<a class="sourceLine" id="cb319-30" title="30">  }</a>
<a class="sourceLine" id="cb319-31" title="31">}</a>
<a class="sourceLine" id="cb319-32" title="32"></a>
<a class="sourceLine" id="cb319-33" title="33">condition_label_fun &lt;-<span class="st"> </span><span class="cf">function</span>(has_sensors, env_chg) {</a>
<a class="sourceLine" id="cb319-34" title="34">  <span class="cf">if</span> (has_sensors <span class="op">&amp;&amp;</span><span class="st"> </span>env_chg) {</a>
<a class="sourceLine" id="cb319-35" title="35">    <span class="kw">return</span>(<span class="st">&quot;PLASTIC&quot;</span>)</a>
<a class="sourceLine" id="cb319-36" title="36">  } <span class="cf">else</span> <span class="cf">if</span> (env_chg) {</a>
<a class="sourceLine" id="cb319-37" title="37">    <span class="kw">return</span>(<span class="st">&quot;NON-PLASTIC&quot;</span>)</a>
<a class="sourceLine" id="cb319-38" title="38">  } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb319-39" title="39">    <span class="kw">return</span>(<span class="st">&quot;STATIC&quot;</span>)</a>
<a class="sourceLine" id="cb319-40" title="40">  }</a>
<a class="sourceLine" id="cb319-41" title="41">}</a>
<a class="sourceLine" id="cb319-42" title="42"></a>
<a class="sourceLine" id="cb319-43" title="43">summary_data<span class="op">$</span>env_label &lt;-<span class="st"> </span><span class="kw">mapply</span>(</a>
<a class="sourceLine" id="cb319-44" title="44">  env_label_fun,</a>
<a class="sourceLine" id="cb319-45" title="45">  summary_data<span class="op">$</span>chg_env</a>
<a class="sourceLine" id="cb319-46" title="46">)</a>
<a class="sourceLine" id="cb319-47" title="47">summary_data<span class="op">$</span>sensors_label &lt;-<span class="st"> </span><span class="kw">mapply</span>(</a>
<a class="sourceLine" id="cb319-48" title="48">  sensors_label_fun,</a>
<a class="sourceLine" id="cb319-49" title="49">  summary_data<span class="op">$</span>sensors</a>
<a class="sourceLine" id="cb319-50" title="50">)</a>
<a class="sourceLine" id="cb319-51" title="51">summary_data<span class="op">$</span>condition &lt;-<span class="st"> </span><span class="kw">mapply</span>(</a>
<a class="sourceLine" id="cb319-52" title="52">  condition_label_fun,</a>
<a class="sourceLine" id="cb319-53" title="53">  summary_data<span class="op">$</span>sensors,</a>
<a class="sourceLine" id="cb319-54" title="54">  summary_data<span class="op">$</span>chg_env</a>
<a class="sourceLine" id="cb319-55" title="55">)</a>
<a class="sourceLine" id="cb319-56" title="56"></a>
<a class="sourceLine" id="cb319-57" title="57">condition_order =<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb319-58" title="58">  <span class="st">&quot;STATIC&quot;</span>,</a>
<a class="sourceLine" id="cb319-59" title="59">  <span class="st">&quot;NON-PLASTIC&quot;</span>,</a>
<a class="sourceLine" id="cb319-60" title="60">  <span class="st">&quot;PLASTIC&quot;</span></a>
<a class="sourceLine" id="cb319-61" title="61">)</a>
<a class="sourceLine" id="cb319-62" title="62"></a>
<a class="sourceLine" id="cb319-63" title="63">pairwise_comparisons &lt;-<span class="st"> </span><span class="kw">list</span>(</a>
<a class="sourceLine" id="cb319-64" title="64">  <span class="kw">c</span>(<span class="st">&quot;STATIC&quot;</span>, <span class="st">&quot;NON-PLASTIC&quot;</span>),</a>
<a class="sourceLine" id="cb319-65" title="65">  <span class="kw">c</span>(<span class="st">&quot;STATIC&quot;</span>, <span class="st">&quot;PLASTIC&quot;</span>),</a>
<a class="sourceLine" id="cb319-66" title="66">  <span class="kw">c</span>(<span class="st">&quot;PLASTIC&quot;</span>, <span class="st">&quot;NON-PLASTIC&quot;</span>)</a>
<a class="sourceLine" id="cb319-67" title="67">)</a>
<a class="sourceLine" id="cb319-68" title="68"></a>
<a class="sourceLine" id="cb319-69" title="69">p_label &lt;-<span class="st"> </span><span class="cf">function</span>(p_value) {</a>
<a class="sourceLine" id="cb319-70" title="70">  threshold =<span class="st"> </span><span class="fl">0.0001</span></a>
<a class="sourceLine" id="cb319-71" title="71">  <span class="cf">if</span> (p_value <span class="op">&lt;</span><span class="st"> </span>threshold) {</a>
<a class="sourceLine" id="cb319-72" title="72">    <span class="kw">return</span>(<span class="kw">paste0</span>(<span class="st">&quot;p &lt; &quot;</span>, threshold))</a>
<a class="sourceLine" id="cb319-73" title="73">  } <span class="cf">else</span> {</a>
<a class="sourceLine" id="cb319-74" title="74">    <span class="kw">return</span>(<span class="kw">paste0</span>(<span class="st">&quot;p = &quot;</span>, p_value))</a>
<a class="sourceLine" id="cb319-75" title="75">  }</a>
<a class="sourceLine" id="cb319-76" title="76">}</a>
<a class="sourceLine" id="cb319-77" title="77"></a>
<a class="sourceLine" id="cb319-78" title="78">poison_penalties &lt;-<span class="st"> </span><span class="kw">levels</span>(summary_data<span class="op">$</span>POISON_PENALTY)</a>
<a class="sourceLine" id="cb319-79" title="79"></a>
<a class="sourceLine" id="cb319-80" title="80"><span class="co">###### time series #####</span></a>
<a class="sourceLine" id="cb319-81" title="81">lineage_time_series_data_loc &lt;-<span class="st"> </span><span class="kw">paste0</span>(working_directory, <span class="st">&quot;data/lineage_series.csv&quot;</span>)</a>
<a class="sourceLine" id="cb319-82" title="82">lineage_time_series_data &lt;-<span class="st"> </span><span class="kw">read.csv</span>(lineage_time_series_data_loc)</a>
<a class="sourceLine" id="cb319-83" title="83"></a>
<a class="sourceLine" id="cb319-84" title="84">lineage_time_series_data<span class="op">$</span>DISABLE_REACTION_SENSORS &lt;-<span class="st"> </span><span class="kw">as.factor</span>(lineage_time_series_data<span class="op">$</span>DISABLE_REACTION_SENSORS)</a>
<a class="sourceLine" id="cb319-85" title="85">lineage_time_series_data<span class="op">$</span>chg_env &lt;-<span class="st"> </span>lineage_time_series_data<span class="op">$</span>chg_env <span class="op">==</span><span class="st"> &quot;True&quot;</span></a>
<a class="sourceLine" id="cb319-86" title="86">lineage_time_series_data<span class="op">$</span>sensors &lt;-<span class="st"> </span>lineage_time_series_data<span class="op">$</span>DISABLE_REACTION_SENSORS <span class="op">==</span><span class="st"> &quot;0&quot;</span></a>
<a class="sourceLine" id="cb319-87" title="87">lineage_time_series_data<span class="op">$</span>POISON_PENALTY &lt;-<span class="st"> </span><span class="kw">as.factor</span>(lineage_time_series_data<span class="op">$</span>POISON_VALUE)</a>
<a class="sourceLine" id="cb319-88" title="88"></a>
<a class="sourceLine" id="cb319-89" title="89">lineage_time_series_data<span class="op">$</span>env_label &lt;-<span class="st"> </span><span class="kw">mapply</span>(</a>
<a class="sourceLine" id="cb319-90" title="90">  env_label_fun,</a>
<a class="sourceLine" id="cb319-91" title="91">  lineage_time_series_data<span class="op">$</span>chg_env</a>
<a class="sourceLine" id="cb319-92" title="92">)</a>
<a class="sourceLine" id="cb319-93" title="93">lineage_time_series_data<span class="op">$</span>sensors_label &lt;-<span class="st"> </span><span class="kw">mapply</span>(</a>
<a class="sourceLine" id="cb319-94" title="94">  sensors_label_fun,</a>
<a class="sourceLine" id="cb319-95" title="95">  lineage_time_series_data<span class="op">$</span>sensors</a>
<a class="sourceLine" id="cb319-96" title="96">)</a>
<a class="sourceLine" id="cb319-97" title="97">lineage_time_series_data<span class="op">$</span>condition &lt;-<span class="st"> </span><span class="kw">mapply</span>(</a>
<a class="sourceLine" id="cb319-98" title="98">  condition_label_fun,</a>
<a class="sourceLine" id="cb319-99" title="99">  lineage_time_series_data<span class="op">$</span>sensors,</a>
<a class="sourceLine" id="cb319-100" title="100">  lineage_time_series_data<span class="op">$</span>chg_env</a>
<a class="sourceLine" id="cb319-101" title="101">)</a>
<a class="sourceLine" id="cb319-102" title="102"></a>
<a class="sourceLine" id="cb319-103" title="103"><span class="co">####### misc #######</span></a>
<a class="sourceLine" id="cb319-104" title="104"><span class="co"># Configure our default graphing theme</span></a>
<a class="sourceLine" id="cb319-105" title="105">focal_summary_data &lt;-<span class="st"> </span><span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>focal_poison_penalty)</a>
<a class="sourceLine" id="cb319-106" title="106"><span class="kw">theme_set</span>(<span class="kw">theme_cowplot</span>())</a>
<a class="sourceLine" id="cb319-107" title="107">cb_palette &lt;-<span class="st"> &quot;Paired&quot;</span></a>
<a class="sourceLine" id="cb319-108" title="108"><span class="kw">dir.create</span>(<span class="kw">paste0</span>(working_directory, <span class="st">&quot;plots&quot;</span>), <span class="dt">showWarnings=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb319-109" title="109">samplemean &lt;-<span class="st"> </span><span class="cf">function</span>(x, d) {</a>
<a class="sourceLine" id="cb319-110" title="110">  <span class="kw">return</span>(<span class="kw">mean</span>(x[d]))</a>
<a class="sourceLine" id="cb319-111" title="111">}</a></code></pre></div>
</div>
<div id="evolution-of-phenotypic-plasticity-1" class="section level2">
<h2><span class="header-section-number">7.4</span> Evolution of phenotypic plasticity</h2>
<p>For sensor-enabled populations in fluctuating environments, we only transfered populations containing an optimally plastic genotype to phase-two.</p>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb320-1" title="1">summary_data_grouped =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(summary_data, sensors, env_label, condition, POISON_PENALTY)</a>
<a class="sourceLine" id="cb320-2" title="2">summary_data_group_counts =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">summarize</span>(summary_data_grouped, <span class="dt">n=</span>dplyr<span class="op">::</span><span class="kw">n</span>())</a>
<a class="sourceLine" id="cb320-3" title="3"></a>
<a class="sourceLine" id="cb320-4" title="4"><span class="kw">ggplot</span>(summary_data_group_counts, <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>n, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-5" title="5"><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-6" title="6"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>n, <span class="dt">y=</span>n<span class="op">+</span><span class="dv">2</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-7" title="7"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb320-8" title="8">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb320-9" title="9">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb320-10" title="10">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-11" title="11"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb320-12" title="12">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb320-13" title="13">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-14" title="14"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb320-15" title="15">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb320-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-17" title="17"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of replicates in phase two&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-18" title="18"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>POISON_PENALTY, <span class="dt">labeller=</span>label_both) <span class="op">+</span></a>
<a class="sourceLine" id="cb320-19" title="19"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb320-20" title="20">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb320-21" title="21">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-72-1.png" width="672" /></p>
<p>We can confirm our expectation that the dominant genotypes in non-plastic conditions are not phenotypically plastic.</p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb321-1" title="1">summary_data_grouped =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(summary_data, condition, is_plastic, POISON_PENALTY)</a>
<a class="sourceLine" id="cb321-2" title="2">summary_data_group_counts =<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">summarize</span>(summary_data_grouped, <span class="dt">n=</span>dplyr<span class="op">::</span><span class="kw">n</span>())</a></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;condition&#39;, &#39;is_plastic&#39;. You can override using the `.groups` argument.</code></pre>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb323-1" title="1"><span class="kw">ggplot</span>(<span class="kw">filter</span>(summary_data_group_counts, is_plastic), <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>n, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-2" title="2"><span class="st">  </span><span class="kw">geom_col</span>(<span class="dt">position=</span><span class="kw">position_dodge</span>(<span class="fl">0.9</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-3" title="3"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb323-4" title="4">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb323-5" title="5">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb323-6" title="6">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-7" title="7"><span class="st">  </span><span class="kw">geom_text</span>(<span class="kw">aes</span>(<span class="dt">label=</span>n, <span class="dt">y=</span>n<span class="op">+</span><span class="dv">1</span>)) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-8" title="8"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb323-9" title="9">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb323-10" title="10">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-11" title="11"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb323-12" title="12">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb323-13" title="13">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-14" title="14"><span class="st">  </span><span class="kw">ylab</span>(<span class="st">&quot;Number of replicates with a plastic dominant genotype&quot;</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-15" title="15"><span class="st">  </span><span class="kw">ylim</span>(<span class="dv">0</span>, <span class="dv">100</span>) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-16" title="16"><span class="st">  </span><span class="kw">facet_wrap</span>(<span class="op">~</span>POISON_PENALTY, <span class="dt">labeller=</span>label_both) <span class="op">+</span></a>
<a class="sourceLine" id="cb323-17" title="17"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb323-18" title="18">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb323-19" title="19">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-73-1.png" width="672" /></p>
</div>
<div id="poison-instruction-execution" class="section level2">
<h2><span class="header-section-number">7.5</span> Poison instruction execution</h2>
<div id="number-of-replicates-where-final-dominant-genotype-executes-the-poison-instruction" class="section level3">
<h3><span class="header-section-number">7.5.1</span> Number of replicates where final dominant genotype executes the poison instruction</h3>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb324-1" title="1"><span class="cf">for</span> (penalty <span class="cf">in</span> poison_penalties) {</a>
<a class="sourceLine" id="cb324-2" title="2">  occurrences &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb324-3" title="3">    <span class="kw">length</span>(<span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;NON-PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>dominant_times_poison_executed <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)<span class="op">$</span>RANDOM_SEED),</a>
<a class="sourceLine" id="cb324-4" title="4">    <span class="kw">length</span>(<span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>dominant_times_poison_executed <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)<span class="op">$</span>RANDOM_SEED),</a>
<a class="sourceLine" id="cb324-5" title="5">    <span class="kw">length</span>(<span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;STATIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>dominant_times_poison_executed <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)<span class="op">$</span>RANDOM_SEED)</a>
<a class="sourceLine" id="cb324-6" title="6">  )</a>
<a class="sourceLine" id="cb324-7" title="7">  trials &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb324-8" title="8">    <span class="kw">length</span>(<span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;NON-PLASTIC&quot;</span>)<span class="op">$</span>RANDOM_SEED),</a>
<a class="sourceLine" id="cb324-9" title="9">    <span class="kw">length</span>(<span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span>)<span class="op">$</span>RANDOM_SEED),</a>
<a class="sourceLine" id="cb324-10" title="10">    <span class="kw">length</span>(<span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;STATIC&quot;</span> )<span class="op">$</span>RANDOM_SEED)</a>
<a class="sourceLine" id="cb324-11" title="11">  )</a>
<a class="sourceLine" id="cb324-12" title="12">  <span class="kw">names</span>(trials) &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb324-13" title="13">    <span class="st">&quot;NON-PLASTIC&quot;</span>,</a>
<a class="sourceLine" id="cb324-14" title="14">    <span class="st">&quot;PLASTIC&quot;</span>,</a>
<a class="sourceLine" id="cb324-15" title="15">    <span class="st">&quot;STATIC&quot;</span></a>
<a class="sourceLine" id="cb324-16" title="16">  )</a>
<a class="sourceLine" id="cb324-17" title="17">  <span class="kw">names</span>(occurrences) &lt;-<span class="st"> </span><span class="kw">c</span>(</a>
<a class="sourceLine" id="cb324-18" title="18">    <span class="st">&quot;NON-PLASTIC&quot;</span>,</a>
<a class="sourceLine" id="cb324-19" title="19">    <span class="st">&quot;PLASTIC&quot;</span>,</a>
<a class="sourceLine" id="cb324-20" title="20">    <span class="st">&quot;STATIC&quot;</span></a>
<a class="sourceLine" id="cb324-21" title="21">  )</a>
<a class="sourceLine" id="cb324-22" title="22">  poison_exec_table &lt;-<span class="st"> </span><span class="kw">data.frame</span>(</a>
<a class="sourceLine" id="cb324-23" title="23">    <span class="dt">executes.poison=</span>occurrences,</a>
<a class="sourceLine" id="cb324-24" title="24">    <span class="dt">replicates=</span>trials</a>
<a class="sourceLine" id="cb324-25" title="25">  )</a>
<a class="sourceLine" id="cb324-26" title="26">  <span class="kw">cat</span>(<span class="kw">paste0</span>(<span class="st">&quot;#### Penalty: &quot;</span>, penalty, <span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>))</a>
<a class="sourceLine" id="cb324-27" title="27">  <span class="kw">cat</span>(<span class="kw">print</span>(<span class="kw">kable</span>(poison_exec_table)))</a>
<a class="sourceLine" id="cb324-28" title="28">  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb324-29" title="29">  ft &lt;-<span class="st"> </span><span class="kw">pairwise.fisher.test</span>(<span class="dt">x=</span>occurrences, <span class="dt">n=</span>trials, <span class="dt">p.adjust.method=</span><span class="st">&quot;bonferroni&quot;</span>)</a>
<a class="sourceLine" id="cb324-30" title="30">  <span class="kw">print</span>(ft)</a>
<a class="sourceLine" id="cb324-31" title="31">  <span class="kw">cat</span>(<span class="st">&quot;</span><span class="ch">\n\n</span><span class="st">&quot;</span>)</a>
<a class="sourceLine" id="cb324-32" title="32">}</a></code></pre></div>
<pre><code>## #### Penalty: 0
## 
## 
## |            | executes.poison| replicates|
## |:-----------|---------------:|----------:|
## |NON-PLASTIC |              86|        100|
## |PLASTIC     |              27|         41|
## |STATIC      |              85|        100|
## 
## 
##  Pairwise comparisons using Pairwise comparison of proportions (Fisher) 
## 
## data:  occurrences out of trials 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 0.03        -      
## STATIC  1.00        0.06   
## 
## P value adjustment method: bonferroni 
## 
## 
## #### Penalty: 0.03
## 
## 
## |            | executes.poison| replicates|
## |:-----------|---------------:|----------:|
## |NON-PLASTIC |              46|        100|
## |PLASTIC     |               1|         42|
## |STATIC      |               1|        100|
## 
## 
##  Pairwise comparisons using Pairwise comparison of proportions (Fisher) 
## 
## data:  occurrences out of trials 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 1.2e-07     -      
## STATIC  2.9e-15     1      
## 
## P value adjustment method: bonferroni 
## 
## 
## #### Penalty: 0.1
## 
## 
## |            | executes.poison| replicates|
## |:-----------|---------------:|----------:|
## |NON-PLASTIC |              14|        100|
## |PLASTIC     |               0|         43|
## |STATIC      |               0|        100|
## 
## 
##  Pairwise comparisons using Pairwise comparison of proportions (Fisher) 
## 
## data:  occurrences out of trials 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 0.03212     -      
## STATIC  0.00022     1.00000
## 
## P value adjustment method: bonferroni 
## 
## 
## #### Penalty: 0.3
## 
## 
## |            | executes.poison| replicates|
## |:-----------|---------------:|----------:|
## |NON-PLASTIC |               0|        100|
## |PLASTIC     |               0|         44|
## |STATIC      |               0|        100|
## 
## 
##  Pairwise comparisons using Pairwise comparison of proportions (Fisher) 
## 
## data:  occurrences out of trials 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 1           -      
## STATIC  1           1      
## 
## P value adjustment method: bonferroni</code></pre>
</div>
<div id="poison-instruction-execution-final-population" class="section level3">
<h3><span class="header-section-number">7.5.2</span> Poison instruction execution (final population)</h3>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb326-1" title="1"><span class="kw">ggplot</span>(summary_data, <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>final_population_poison, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-2" title="2"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb326-3" title="3">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb326-4" title="4">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb326-5" title="5">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb326-7" title="7">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb326-8" title="8">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb326-9" title="9">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb326-10" title="10">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb326-11" title="11">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-12" title="12"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb326-13" title="13">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb326-14" title="14">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb326-15" title="15">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb326-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-17" title="17"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb326-18" title="18">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb326-19" title="19">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb326-20" title="20">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-21" title="21"><span class="st">  </span><span class="kw">scale_y_continuous</span>(</a>
<a class="sourceLine" id="cb326-22" title="22">    <span class="dt">name=</span><span class="st">&quot;poison instruction executions (final population)&quot;</span>,</a>
<a class="sourceLine" id="cb326-23" title="23">    <span class="dt">trans=</span><span class="kw">pseudo_log_trans</span>(<span class="dt">sigma=</span><span class="dv">1</span>,<span class="dt">base=</span><span class="dv">10</span>),</a>
<a class="sourceLine" id="cb326-24" title="24">    <span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">100</span>,<span class="dv">10000</span>,<span class="dv">1000000</span>),</a>
<a class="sourceLine" id="cb326-25" title="25">    <span class="dt">limits=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">1000000</span>)</a>
<a class="sourceLine" id="cb326-26" title="26">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-27" title="27"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb326-28" title="28">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb326-29" title="29">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-30" title="30"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb326-31" title="31">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb326-32" title="32">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-33" title="33"><span class="st">  </span><span class="kw">facet_wrap</span>(</a>
<a class="sourceLine" id="cb326-34" title="34">    <span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb326-35" title="35">    <span class="dt">labeller=</span>label_both</a>
<a class="sourceLine" id="cb326-36" title="36">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-37" title="37"><span class="st">  </span><span class="co"># coord_flip() +</span></a>
<a class="sourceLine" id="cb326-38" title="38"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb326-39" title="39">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb326-40" title="40">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb326-41" title="41"><span class="st">  </span><span class="kw">ggsave</span>(</a>
<a class="sourceLine" id="cb326-42" title="42">    <span class="kw">paste0</span>(working_directory, <span class="st">&quot;plots/final-population-poison-log.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb326-43" title="43">    <span class="dt">width=</span><span class="dv">15</span>,</a>
<a class="sourceLine" id="cb326-44" title="44">    <span class="dt">height=</span><span class="dv">10</span></a>
<a class="sourceLine" id="cb326-45" title="45">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb327-1" title="1"><span class="cf">for</span> (penalty <span class="cf">in</span> poison_penalties) {</a>
<a class="sourceLine" id="cb327-2" title="2">  stat_data &lt;-<span class="st"> </span><span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty)</a>
<a class="sourceLine" id="cb327-3" title="3">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb327-4" title="4">    <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb327-5" title="5">      <span class="st">&quot;PENALTY: &quot;</span>, penalty</a>
<a class="sourceLine" id="cb327-6" title="6">    )</a>
<a class="sourceLine" id="cb327-7" title="7">  )</a>
<a class="sourceLine" id="cb327-8" title="8">  kt &lt;-<span class="st"> </span><span class="kw">kruskal.test</span>(</a>
<a class="sourceLine" id="cb327-9" title="9">      <span class="dt">formula=</span>final_population_poison<span class="op">~</span>condition,</a>
<a class="sourceLine" id="cb327-10" title="10">      <span class="dt">data=</span>stat_data</a>
<a class="sourceLine" id="cb327-11" title="11">    )</a>
<a class="sourceLine" id="cb327-12" title="12">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb327-13" title="13">    kt</a>
<a class="sourceLine" id="cb327-14" title="14">  )</a>
<a class="sourceLine" id="cb327-15" title="15">  <span class="cf">if</span> (<span class="kw">is.na</span>(kt<span class="op">$</span>p.value)) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb327-16" title="16">  <span class="cf">if</span> (kt<span class="op">$</span>p.value <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb327-17" title="17">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb327-18" title="18">    <span class="kw">pairwise.wilcox.test</span>(</a>
<a class="sourceLine" id="cb327-19" title="19">      <span class="dt">x=</span>stat_data<span class="op">$</span>final_population_poison,</a>
<a class="sourceLine" id="cb327-20" title="20">      <span class="dt">g=</span>stat_data<span class="op">$</span>condition,</a>
<a class="sourceLine" id="cb327-21" title="21">      <span class="dt">p.adjust.method=</span><span class="st">&quot;bonferroni&quot;</span></a>
<a class="sourceLine" id="cb327-22" title="22">    )</a>
<a class="sourceLine" id="cb327-23" title="23">  )</a>
<a class="sourceLine" id="cb327-24" title="24">}</a></code></pre></div>
<pre><code>## [1] &quot;PENALTY: 0&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  final_population_poison by condition
## Kruskal-Wallis chi-squared = 43.589, df = 2, p-value = 3.426e-10
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$final_population_poison and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 8.7e-07     -      
## STATIC  9.8e-07     0.00074
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.03&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  final_population_poison by condition
## Kruskal-Wallis chi-squared = 20.74, df = 2, p-value = 3.136e-05
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$final_population_poison and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 0.003       -      
## STATIC  1e-04       1.000  
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.1&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  final_population_poison by condition
## Kruskal-Wallis chi-squared = 20.608, df = 2, p-value = 3.35e-05
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$final_population_poison and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 0.0093      -      
## STATIC  4.9e-05     1.0000 
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.3&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  final_population_poison by condition
## Kruskal-Wallis chi-squared = 3.3994, df = 2, p-value = 0.1827</code></pre>
</div>
<div id="cummulative-poison-instruction-execution-along-final-dominant-lineages" class="section level3">
<h3><span class="header-section-number">7.5.3</span> Cummulative poison instruction execution along final dominant lineages</h3>
<div class="sourceCode" id="cb329"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb329-1" title="1"><span class="kw">ggplot</span>(summary_data, <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>dominant_lineage_times_poison_executed, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-2" title="2"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb329-3" title="3">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb329-4" title="4">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb329-5" title="5">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb329-7" title="7">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb329-8" title="8">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb329-9" title="9">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb329-10" title="10">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb329-11" title="11">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-12" title="12"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb329-13" title="13">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb329-14" title="14">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb329-15" title="15">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb329-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-17" title="17"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb329-18" title="18">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb329-19" title="19">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb329-20" title="20">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-21" title="21"><span class="st">  </span><span class="kw">scale_y_continuous</span>(</a>
<a class="sourceLine" id="cb329-22" title="22">    <span class="dt">name=</span><span class="st">&quot;poison instruction executions (dominant lineage)&quot;</span>,</a>
<a class="sourceLine" id="cb329-23" title="23">    <span class="dt">trans=</span><span class="kw">pseudo_log_trans</span>(<span class="dt">sigma =</span> <span class="dv">1</span>, <span class="dt">base =</span> <span class="dv">10</span>),</a>
<a class="sourceLine" id="cb329-24" title="24">    <span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">1000</span>,<span class="dv">100000</span>),</a>
<a class="sourceLine" id="cb329-25" title="25">    <span class="dt">limits=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>,<span class="dv">100000</span>)</a>
<a class="sourceLine" id="cb329-26" title="26">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-27" title="27"><span class="st">  </span><span class="kw">facet_wrap</span>(</a>
<a class="sourceLine" id="cb329-28" title="28">    <span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb329-29" title="29">    <span class="dt">labeller=</span>label_both</a>
<a class="sourceLine" id="cb329-30" title="30">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-31" title="31"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb329-32" title="32">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb329-33" title="33">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-34" title="34"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb329-35" title="35">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb329-36" title="36">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-37" title="37"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb329-38" title="38">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb329-39" title="39">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb329-40" title="40"><span class="st">  </span><span class="kw">ggsave</span>(</a>
<a class="sourceLine" id="cb329-41" title="41">    <span class="kw">paste0</span>(working_directory, <span class="st">&quot;plots/final-dominant-lineage-poison-log.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb329-42" title="42">    <span class="dt">width=</span><span class="dv">15</span>,</a>
<a class="sourceLine" id="cb329-43" title="43">    <span class="dt">height=</span><span class="dv">10</span></a>
<a class="sourceLine" id="cb329-44" title="44">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-77-1.png" width="672" /></p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb330-1" title="1"><span class="cf">for</span> (penalty <span class="cf">in</span> poison_penalties) {</a>
<a class="sourceLine" id="cb330-2" title="2">  stat_data &lt;-<span class="st"> </span><span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty)</a>
<a class="sourceLine" id="cb330-3" title="3">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb330-4" title="4">    <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb330-5" title="5">      <span class="st">&quot;PENALTY: &quot;</span>, penalty</a>
<a class="sourceLine" id="cb330-6" title="6">    )</a>
<a class="sourceLine" id="cb330-7" title="7">  )</a>
<a class="sourceLine" id="cb330-8" title="8">  kt &lt;-<span class="st"> </span><span class="kw">kruskal.test</span>(</a>
<a class="sourceLine" id="cb330-9" title="9">      <span class="dt">formula=</span>dominant_lineage_times_poison_executed<span class="op">~</span>condition,</a>
<a class="sourceLine" id="cb330-10" title="10">      <span class="dt">data=</span>stat_data</a>
<a class="sourceLine" id="cb330-11" title="11">    )</a>
<a class="sourceLine" id="cb330-12" title="12">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb330-13" title="13">    kt</a>
<a class="sourceLine" id="cb330-14" title="14">  )</a>
<a class="sourceLine" id="cb330-15" title="15">  <span class="cf">if</span> (<span class="kw">is.na</span>(kt<span class="op">$</span>p.value)) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb330-16" title="16">  <span class="cf">if</span> (kt<span class="op">$</span>p.value <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb330-17" title="17">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb330-18" title="18">    <span class="kw">pairwise.wilcox.test</span>(</a>
<a class="sourceLine" id="cb330-19" title="19">      <span class="dt">x=</span>stat_data<span class="op">$</span>dominant_lineage_times_poison_executed,</a>
<a class="sourceLine" id="cb330-20" title="20">      <span class="dt">g=</span>stat_data<span class="op">$</span>condition,</a>
<a class="sourceLine" id="cb330-21" title="21">      <span class="dt">p.adjust.method=</span><span class="st">&quot;bonferroni&quot;</span></a>
<a class="sourceLine" id="cb330-22" title="22">    )</a>
<a class="sourceLine" id="cb330-23" title="23">  )</a>
<a class="sourceLine" id="cb330-24" title="24">}</a></code></pre></div>
<pre><code>## [1] &quot;PENALTY: 0&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_times_poison_executed by condition
## Kruskal-Wallis chi-squared = 178.84, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_times_poison_executed and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.0018 
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.03&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_times_poison_executed by condition
## Kruskal-Wallis chi-squared = 178.62, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_times_poison_executed and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.011  
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.1&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_times_poison_executed by condition
## Kruskal-Wallis chi-squared = 184.83, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_times_poison_executed and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.21   
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.3&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_times_poison_executed by condition
## Kruskal-Wallis chi-squared = 149.48, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_times_poison_executed and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 4.4e-16     -      
## STATIC  &lt; 2e-16     0.84   
## 
## P value adjustment method: bonferroni</code></pre>
</div>
</div>
<div id="characterizing-mutations-that-increase-poison-instruction-execution" class="section level2">
<h2><span class="header-section-number">7.6</span> Characterizing mutations that increase poison instruction execution</h2>
<div id="number-of-offspring-along-dominant-lineage-with-increase-in-poison-instruction-execution" class="section level3">
<h3><span class="header-section-number">7.6.1</span> Number of offspring along dominant lineage with increase in poison instruction execution</h3>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb332-1" title="1"><span class="kw">ggplot</span>(summary_data, <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-2" title="2"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb332-3" title="3">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb332-4" title="4">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb332-5" title="5">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb332-7" title="7">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb332-8" title="8">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb332-9" title="9">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb332-10" title="10">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb332-11" title="11">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-12" title="12"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb332-13" title="13">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb332-14" title="14">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb332-15" title="15">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb332-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-17" title="17"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb332-18" title="18">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb332-19" title="19">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb332-20" title="20">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-21" title="21"><span class="st">  </span><span class="kw">facet_wrap</span>(</a>
<a class="sourceLine" id="cb332-22" title="22">    <span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb332-23" title="23">    <span class="dt">labeller=</span>label_both</a>
<a class="sourceLine" id="cb332-24" title="24">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-25" title="25"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb332-26" title="26">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb332-27" title="27">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-28" title="28"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb332-29" title="29">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb332-30" title="30">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb332-31" title="31"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb332-32" title="32">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb332-33" title="33">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-79-1.png" width="672" /></p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb333-1" title="1"><span class="cf">for</span> (penalty <span class="cf">in</span> poison_penalties) {</a>
<a class="sourceLine" id="cb333-2" title="2">  stat_data &lt;-<span class="st"> </span><span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty)</a>
<a class="sourceLine" id="cb333-3" title="3">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb333-4" title="4">    <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb333-5" title="5">      <span class="st">&quot;PENALTY: &quot;</span>, penalty</a>
<a class="sourceLine" id="cb333-6" title="6">    )</a>
<a class="sourceLine" id="cb333-7" title="7">  )</a>
<a class="sourceLine" id="cb333-8" title="8">  kt &lt;-<span class="st"> </span><span class="kw">kruskal.test</span>(</a>
<a class="sourceLine" id="cb333-9" title="9">      <span class="dt">formula=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">~</span>condition,</a>
<a class="sourceLine" id="cb333-10" title="10">      <span class="dt">data=</span>stat_data</a>
<a class="sourceLine" id="cb333-11" title="11">    )</a>
<a class="sourceLine" id="cb333-12" title="12">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb333-13" title="13">    kt</a>
<a class="sourceLine" id="cb333-14" title="14">  )</a>
<a class="sourceLine" id="cb333-15" title="15">  <span class="cf">if</span> (<span class="kw">is.na</span>(kt<span class="op">$</span>p.value)) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb333-16" title="16">  <span class="cf">if</span> (kt<span class="op">$</span>p.value <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb333-17" title="17">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb333-18" title="18">    <span class="kw">pairwise.wilcox.test</span>(</a>
<a class="sourceLine" id="cb333-19" title="19">      <span class="dt">x=</span>stat_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases,</a>
<a class="sourceLine" id="cb333-20" title="20">      <span class="dt">g=</span>stat_data<span class="op">$</span>condition,</a>
<a class="sourceLine" id="cb333-21" title="21">      <span class="dt">p.adjust.method=</span><span class="st">&quot;bonferroni&quot;</span></a>
<a class="sourceLine" id="cb333-22" title="22">    )</a>
<a class="sourceLine" id="cb333-23" title="23">  )</a>
<a class="sourceLine" id="cb333-24" title="24">}</a></code></pre></div>
<pre><code>## [1] &quot;PENALTY: 0&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases by condition
## Kruskal-Wallis chi-squared = 179.79, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt; 2e-16     -      
## STATIC  &lt; 2e-16     0.00046
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.03&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases by condition
## Kruskal-Wallis chi-squared = 179.35, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.03   
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.1&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases by condition
## Kruskal-Wallis chi-squared = 185.34, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.27   
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.3&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases by condition
## Kruskal-Wallis chi-squared = 146.35, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 7.8e-16     -      
## STATIC  &lt; 2e-16     0.86   
## 
## P value adjustment method: bonferroni</code></pre>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb335-1" title="1"><span class="co"># sum(filter(summary_data, condition==&quot;NON-PLASTIC&quot; &amp; POISON_PENALTY==0.1)$dominant_lineage_num_times_hitchhike_inst_exec_increases)</span></a>
<a class="sourceLine" id="cb335-2" title="2"><span class="co"># sum(filter(summary_data, condition==&quot;PLASTIC&quot; &amp; POISON_PENALTY==0.1)$dominant_lineage_num_times_hitchhike_inst_exec_increases)</span></a>
<a class="sourceLine" id="cb335-3" title="3"><span class="co"># sum(filter(summary_data, condition==&quot;STATIC&quot; &amp; POISON_PENALTY==0.1)$dominant_lineage_num_times_hitchhike_inst_exec_increases)</span></a></code></pre></div>
<p>Focal figure for the manuscript:</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb336-1" title="1"><span class="co"># Compute manual labels for geom_signif</span></a>
<a class="sourceLine" id="cb336-2" title="2">stat.test &lt;-<span class="st"> </span>focal_summary_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb336-3" title="3"><span class="st">  </span><span class="kw">wilcox_test</span>(dominant_lineage_num_times_hitchhike_inst_exec_increases <span class="op">~</span><span class="st"> </span>condition) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb336-4" title="4"><span class="st">  </span><span class="kw">adjust_pvalue</span>(<span class="dt">method =</span> <span class="st">&quot;bonferroni&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb336-5" title="5"><span class="st">  </span><span class="kw">add_significance</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb336-6" title="6"><span class="st">  </span><span class="kw">add_xy_position</span>(<span class="dt">x=</span><span class="st">&quot;condition&quot;</span>)</a>
<a class="sourceLine" id="cb336-7" title="7"><span class="co"># Tweak y.position manually to account for scaled axis (edge case that triggers bad behavior in geom_signif)</span></a>
<a class="sourceLine" id="cb336-8" title="8">stat.test<span class="op">$</span>manual_position &lt;-<span class="st">  </span>stat.test<span class="op">$</span>y.position</a>
<a class="sourceLine" id="cb336-9" title="9">stat.test<span class="op">$</span>label &lt;-<span class="st"> </span><span class="kw">mapply</span>(p_label,stat.test<span class="op">$</span>p.adj)</a>
<a class="sourceLine" id="cb336-10" title="10"></a>
<a class="sourceLine" id="cb336-11" title="11">poison_increases_fig &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb336-12" title="12">    focal_summary_data,</a>
<a class="sourceLine" id="cb336-13" title="13">    <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases, <span class="dt">fill=</span>condition)</a>
<a class="sourceLine" id="cb336-14" title="14">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-15" title="15"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb336-16" title="16">    <span class="dt">scale=</span><span class="st">&quot;width&quot;</span>,</a>
<a class="sourceLine" id="cb336-17" title="17">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb336-18" title="18">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb336-19" title="19">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-20" title="20"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb336-21" title="21">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb336-22" title="22">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb336-23" title="23">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb336-24" title="24">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb336-25" title="25">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-26" title="26"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb336-27" title="27">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb336-28" title="28">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb336-29" title="29">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb336-30" title="30">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-31" title="31"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb336-32" title="32">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb336-33" title="33">    <span class="dt">limits=</span>condition_order,</a>
<a class="sourceLine" id="cb336-34" title="34">    <span class="dt">labels=</span>condition_order</a>
<a class="sourceLine" id="cb336-35" title="35">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-36" title="36"><span class="st">  </span><span class="kw">scale_y_continuous</span>(</a>
<a class="sourceLine" id="cb336-37" title="37">    <span class="dt">name=</span><span class="st">&quot;Poisonous task acquisition count&quot;</span>,</a>
<a class="sourceLine" id="cb336-38" title="38">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-39" title="39"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb336-40" title="40">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb336-41" title="41">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-42" title="42"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb336-43" title="43">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb336-44" title="44">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-45" title="45"><span class="st">  </span><span class="co"># coord_flip()</span></a>
<a class="sourceLine" id="cb336-46" title="46"><span class="st">  </span><span class="kw">labs</span>(</a>
<a class="sourceLine" id="cb336-47" title="47">    <span class="dt">subtitle=</span><span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb336-48" title="48">      <span class="st">&quot;Kruskal-Wallis, &quot;</span>,</a>
<a class="sourceLine" id="cb336-49" title="49">      <span class="kw">p_label</span>(<span class="kw">signif</span>(<span class="kw">kruskal.test</span>(<span class="dt">formula=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">~</span>condition, <span class="dt">data=</span>focal_summary_data)<span class="op">$</span>p.value,<span class="dt">digits=</span><span class="dv">4</span>))</a>
<a class="sourceLine" id="cb336-50" title="50">    )</a>
<a class="sourceLine" id="cb336-51" title="51">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-52" title="52"><span class="st">  </span>ggsignif<span class="op">::</span><span class="kw">geom_signif</span>(</a>
<a class="sourceLine" id="cb336-53" title="53">    <span class="dt">data=</span><span class="kw">filter</span>(stat.test, p.adj <span class="op">&lt;=</span><span class="st"> </span>alpha),</a>
<a class="sourceLine" id="cb336-54" title="54">    <span class="kw">aes</span>(<span class="dt">xmin=</span>group1,<span class="dt">xmax=</span>group2,<span class="dt">annotations=</span>label,<span class="dt">y_position=</span>manual_position),</a>
<a class="sourceLine" id="cb336-55" title="55">    <span class="dt">manual=</span><span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb336-56" title="56">    <span class="dt">inherit.aes=</span><span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb336-57" title="57">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb336-58" title="58"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb336-59" title="59">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb336-60" title="60">  )</a></code></pre></div>
<pre><code>## Warning: Ignoring unknown aesthetics: xmin, xmax, annotations, y_position</code></pre>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb338-1" title="1">poison_increases_fig</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
</div>
<div id="frequency-of-increases-in-poison-instruction-execution-lineage" class="section level3">
<h3><span class="header-section-number">7.6.2</span> Frequency of increases in poison instruction execution (lineage)</h3>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb339-1" title="1"><span class="kw">ggplot</span>(summary_data, <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-2" title="2"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb339-3" title="3">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb339-4" title="4">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb339-5" title="5">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb339-7" title="7">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb339-8" title="8">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb339-9" title="9">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb339-10" title="10">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb339-11" title="11">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-12" title="12"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb339-13" title="13">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb339-14" title="14">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb339-15" title="15">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb339-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-17" title="17"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb339-18" title="18">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb339-19" title="19">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb339-20" title="20">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-21" title="21"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb339-22" title="22">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb339-23" title="23">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-24" title="24"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb339-25" title="25">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb339-26" title="26">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-27" title="27"><span class="st">  </span><span class="kw">facet_wrap</span>(</a>
<a class="sourceLine" id="cb339-28" title="28">    <span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb339-29" title="29">    <span class="dt">labeller=</span>label_both,</a>
<a class="sourceLine" id="cb339-30" title="30">    <span class="dt">scales=</span><span class="st">&quot;free_y&quot;</span></a>
<a class="sourceLine" id="cb339-31" title="31">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-32" title="32"><span class="st">  </span><span class="co"># coord_flip() +</span></a>
<a class="sourceLine" id="cb339-33" title="33"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb339-34" title="34">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb339-35" title="35">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb339-36" title="36"><span class="st">  </span><span class="kw">ggsave</span>(</a>
<a class="sourceLine" id="cb339-37" title="37">    <span class="kw">paste0</span>(working_directory, <span class="st">&quot;plots/final-dominant-lineage-poison-increase-per-generation.png&quot;</span>),</a>
<a class="sourceLine" id="cb339-38" title="38">    <span class="dt">width=</span><span class="dv">15</span>,</a>
<a class="sourceLine" id="cb339-39" title="39">    <span class="dt">height=</span><span class="dv">10</span></a>
<a class="sourceLine" id="cb339-40" title="40">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb340-1" title="1"><span class="cf">for</span> (penalty <span class="cf">in</span> poison_penalties) {</a>
<a class="sourceLine" id="cb340-2" title="2">  stat_data &lt;-<span class="st"> </span><span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty)</a>
<a class="sourceLine" id="cb340-3" title="3">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb340-4" title="4">    <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb340-5" title="5">      <span class="st">&quot;PENALTY: &quot;</span>, penalty</a>
<a class="sourceLine" id="cb340-6" title="6">    )</a>
<a class="sourceLine" id="cb340-7" title="7">  )</a>
<a class="sourceLine" id="cb340-8" title="8">  kt &lt;-<span class="st"> </span><span class="kw">kruskal.test</span>(</a>
<a class="sourceLine" id="cb340-9" title="9">      <span class="dt">formula=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation<span class="op">~</span>condition,</a>
<a class="sourceLine" id="cb340-10" title="10">      <span class="dt">data=</span>stat_data</a>
<a class="sourceLine" id="cb340-11" title="11">    )</a>
<a class="sourceLine" id="cb340-12" title="12">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb340-13" title="13">    kt</a>
<a class="sourceLine" id="cb340-14" title="14">  )</a>
<a class="sourceLine" id="cb340-15" title="15">  <span class="cf">if</span> (<span class="kw">is.na</span>(kt<span class="op">$</span>p.value)) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb340-16" title="16">  <span class="cf">if</span> (kt<span class="op">$</span>p.value <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb340-17" title="17">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb340-18" title="18">    <span class="kw">pairwise.wilcox.test</span>(</a>
<a class="sourceLine" id="cb340-19" title="19">      <span class="dt">x=</span>stat_data<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation,</a>
<a class="sourceLine" id="cb340-20" title="20">      <span class="dt">g=</span>stat_data<span class="op">$</span>condition,</a>
<a class="sourceLine" id="cb340-21" title="21">      <span class="dt">p.adjust.method=</span><span class="st">&quot;bonferroni&quot;</span></a>
<a class="sourceLine" id="cb340-22" title="22">    )</a>
<a class="sourceLine" id="cb340-23" title="23">  )</a>
<a class="sourceLine" id="cb340-24" title="24">}</a></code></pre></div>
<pre><code>## [1] &quot;PENALTY: 0&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation by condition
## Kruskal-Wallis chi-squared = 180.05, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt; 2e-16     -      
## STATIC  &lt; 2e-16     7.8e-05
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.03&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation by condition
## Kruskal-Wallis chi-squared = 176.25, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.019  
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.1&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation by condition
## Kruskal-Wallis chi-squared = 184.17, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.2    
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.3&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation by condition
## Kruskal-Wallis chi-squared = 140.99, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 2.2e-15     -      
## STATIC  &lt; 2e-16     0.79   
## 
## P value adjustment method: bonferroni</code></pre>
<p>Figure for the manuscript:</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb342-1" title="1"><span class="co"># Compute manual labels for geom_signif</span></a>
<a class="sourceLine" id="cb342-2" title="2">stat.test &lt;-<span class="st"> </span>focal_summary_data <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb342-3" title="3"><span class="st">  </span><span class="kw">wilcox_test</span>(dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation <span class="op">~</span><span class="st"> </span>condition) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb342-4" title="4"><span class="st">  </span><span class="kw">adjust_pvalue</span>(<span class="dt">method =</span> <span class="st">&quot;bonferroni&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb342-5" title="5"><span class="st">  </span><span class="kw">add_significance</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb342-6" title="6"><span class="st">  </span><span class="kw">add_xy_position</span>(<span class="dt">x=</span><span class="st">&quot;condition&quot;</span>, <span class="dt">step.increase=</span><span class="fl">0.2</span>)</a>
<a class="sourceLine" id="cb342-7" title="7"><span class="co"># Tweak y.position manually to account for scaled axis (edge case that triggers bad behavior in geom_signif)</span></a>
<a class="sourceLine" id="cb342-8" title="8">stat.test<span class="op">$</span>manual_position &lt;-<span class="st">  </span>stat.test<span class="op">$</span>y.position</a>
<a class="sourceLine" id="cb342-9" title="9">stat.test<span class="op">$</span>label &lt;-<span class="st"> </span><span class="kw">mapply</span>(p_label,stat.test<span class="op">$</span>p.adj)</a>
<a class="sourceLine" id="cb342-10" title="10"></a>
<a class="sourceLine" id="cb342-11" title="11">poison_increases_per_gen_fig &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb342-12" title="12">    focal_summary_data,</a>
<a class="sourceLine" id="cb342-13" title="13">    <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation, <span class="dt">fill=</span>condition)</a>
<a class="sourceLine" id="cb342-14" title="14">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-15" title="15"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb342-16" title="16">    <span class="dt">scale=</span><span class="st">&quot;width&quot;</span>,</a>
<a class="sourceLine" id="cb342-17" title="17">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb342-18" title="18">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb342-19" title="19">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-20" title="20"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb342-21" title="21">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb342-22" title="22">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb342-23" title="23">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb342-24" title="24">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb342-25" title="25">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-26" title="26"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb342-27" title="27">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb342-28" title="28">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb342-29" title="29">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb342-30" title="30">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-31" title="31"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb342-32" title="32">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb342-33" title="33">    <span class="dt">limits=</span>condition_order,</a>
<a class="sourceLine" id="cb342-34" title="34">    <span class="dt">labels=</span>condition_order</a>
<a class="sourceLine" id="cb342-35" title="35">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-36" title="36"><span class="st">  </span><span class="kw">scale_y_continuous</span>(</a>
<a class="sourceLine" id="cb342-37" title="37">    <span class="dt">name=</span><span class="st">&quot;Poisonous task acquisition frequency&quot;</span>,</a>
<a class="sourceLine" id="cb342-38" title="38">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-39" title="39"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb342-40" title="40">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb342-41" title="41">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-42" title="42"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb342-43" title="43">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb342-44" title="44">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-45" title="45"><span class="st">  </span><span class="co"># coord_flip()</span></a>
<a class="sourceLine" id="cb342-46" title="46"><span class="st">  </span><span class="kw">labs</span>(</a>
<a class="sourceLine" id="cb342-47" title="47">    <span class="dt">subtitle=</span><span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb342-48" title="48">      <span class="st">&quot;Kruskal-Wallis, &quot;</span>,</a>
<a class="sourceLine" id="cb342-49" title="49">      <span class="kw">p_label</span>(<span class="kw">signif</span>(<span class="kw">kruskal.test</span>(<span class="dt">formula=</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_per_generation<span class="op">~</span>condition, <span class="dt">data=</span>focal_summary_data)<span class="op">$</span>p.value,<span class="dt">digits=</span><span class="dv">4</span>))</a>
<a class="sourceLine" id="cb342-50" title="50">    )</a>
<a class="sourceLine" id="cb342-51" title="51">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-52" title="52"><span class="st">  </span>ggsignif<span class="op">::</span><span class="kw">geom_signif</span>(</a>
<a class="sourceLine" id="cb342-53" title="53">    <span class="dt">data=</span><span class="kw">filter</span>(stat.test, p.adj <span class="op">&lt;=</span><span class="st"> </span>alpha),</a>
<a class="sourceLine" id="cb342-54" title="54">    <span class="kw">aes</span>(<span class="dt">xmin=</span>group1,<span class="dt">xmax=</span>group2,<span class="dt">annotations=</span>label,<span class="dt">y_position=</span>manual_position),</a>
<a class="sourceLine" id="cb342-55" title="55">    <span class="dt">manual=</span><span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb342-56" title="56">    <span class="dt">inherit.aes=</span><span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb342-57" title="57">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb342-58" title="58"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb342-59" title="59">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb342-60" title="60">  )</a></code></pre></div>
<pre><code>## Warning: Ignoring unknown aesthetics: xmin, xmax, annotations, y_position</code></pre>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb344-1" title="1">poison_increases_per_gen_fig</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-84-1.png" width="672" /></p>
</div>
<div id="what-fraction-of-mutations-that-increase-poison-instruction-execution-co-occur-with-base-trait-changes" class="section level3">
<h3><span class="header-section-number">7.6.3</span> What fraction of mutations that increase poison instruction execution co-occur with base trait changes?</h3>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb345-1" title="1"><span class="kw">ggplot</span>(<span class="kw">filter</span>(summary_data, dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">&gt;</span><span class="dv">0</span>), <span class="kw">aes</span>(<span class="dt">x=</span>frac_hitchhiking_linked_trait_change, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb345-2" title="2"><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb345-3" title="3"><span class="st">  </span><span class="kw">facet_grid</span>(</a>
<a class="sourceLine" id="cb345-4" title="4">    condition<span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb345-5" title="5">    <span class="dt">labeller=</span>label_both,</a>
<a class="sourceLine" id="cb345-6" title="6">    <span class="dt">scales=</span><span class="st">&quot;free_y&quot;</span></a>
<a class="sourceLine" id="cb345-7" title="7">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb345-8" title="8"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb345-9" title="9">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb345-10" title="10">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb345-11" title="11"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb345-12" title="12">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb345-13" title="13">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb345-14" title="14"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb345-15" title="15">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb345-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb345-17" title="17"><span class="st">  </span><span class="kw">ggsave</span>(</a>
<a class="sourceLine" id="cb345-18" title="18">    <span class="kw">paste0</span>(working_directory, <span class="st">&quot;plots/dominant-lineage-frac_hitchhiking_linked_trait_change.png&quot;</span>),</a>
<a class="sourceLine" id="cb345-19" title="19">    <span class="dt">width=</span><span class="dv">15</span>,</a>
<a class="sourceLine" id="cb345-20" title="20">    <span class="dt">height=</span><span class="dv">10</span></a>
<a class="sourceLine" id="cb345-21" title="21">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-85-1.png" width="672" /></p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb346-1" title="1"><span class="kw">ggplot</span>(<span class="kw">filter</span>(summary_data, dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">&gt;</span><span class="dv">0</span> ), <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>frac_hitchhiking_linked_trait_change, <span class="dt">fill=</span>condition)) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-2" title="2"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb346-3" title="3">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb346-4" title="4">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb346-5" title="5">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-6" title="6"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb346-7" title="7">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb346-8" title="8">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb346-9" title="9">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb346-10" title="10">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb346-11" title="11">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-12" title="12"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb346-13" title="13">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb346-14" title="14">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb346-15" title="15">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb346-16" title="16">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-17" title="17"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb346-18" title="18">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb346-19" title="19">    <span class="dt">limits=</span>condition_order</a>
<a class="sourceLine" id="cb346-20" title="20">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-21" title="21"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb346-22" title="22">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb346-23" title="23">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-24" title="24"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb346-25" title="25">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb346-26" title="26">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-27" title="27"><span class="st">  </span><span class="kw">facet_wrap</span>(</a>
<a class="sourceLine" id="cb346-28" title="28">    <span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb346-29" title="29">    <span class="dt">labeller=</span>label_both,</a>
<a class="sourceLine" id="cb346-30" title="30">    <span class="dt">scales=</span><span class="st">&quot;free_y&quot;</span></a>
<a class="sourceLine" id="cb346-31" title="31">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb346-32" title="32"><span class="st">  </span><span class="co"># coord_flip() +</span></a>
<a class="sourceLine" id="cb346-33" title="33"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb346-34" title="34">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb346-35" title="35">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-86-1.png" width="672" /></p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb347-1" title="1"><span class="cf">for</span> (penalty <span class="cf">in</span> poison_penalties) {</a>
<a class="sourceLine" id="cb347-2" title="2">  stat_data &lt;-<span class="st"> </span><span class="kw">filter</span>(summary_data, POISON_PENALTY<span class="op">==</span>penalty <span class="op">&amp;</span><span class="st"> </span>dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">&gt;</span><span class="dv">0</span>)</a>
<a class="sourceLine" id="cb347-3" title="3">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb347-4" title="4">    <span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb347-5" title="5">      <span class="st">&quot;PENALTY: &quot;</span>, penalty</a>
<a class="sourceLine" id="cb347-6" title="6">    )</a>
<a class="sourceLine" id="cb347-7" title="7">  )</a>
<a class="sourceLine" id="cb347-8" title="8">  kt &lt;-<span class="st"> </span><span class="kw">kruskal.test</span>(</a>
<a class="sourceLine" id="cb347-9" title="9">      <span class="dt">formula=</span>frac_hitchhiking_linked_trait_change<span class="op">~</span>condition,</a>
<a class="sourceLine" id="cb347-10" title="10">      <span class="dt">data=</span>stat_data</a>
<a class="sourceLine" id="cb347-11" title="11">    )</a>
<a class="sourceLine" id="cb347-12" title="12">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb347-13" title="13">    kt</a>
<a class="sourceLine" id="cb347-14" title="14">  )</a>
<a class="sourceLine" id="cb347-15" title="15">  <span class="cf">if</span> (<span class="kw">is.na</span>(kt<span class="op">$</span>p.value)) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb347-16" title="16">  <span class="cf">if</span> (kt<span class="op">$</span>p.value <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.05</span>) { <span class="cf">next</span> }</a>
<a class="sourceLine" id="cb347-17" title="17">  <span class="kw">print</span>(</a>
<a class="sourceLine" id="cb347-18" title="18">    <span class="kw">pairwise.wilcox.test</span>(</a>
<a class="sourceLine" id="cb347-19" title="19">      <span class="dt">x=</span>stat_data<span class="op">$</span>frac_hitchhiking_linked_trait_change,</a>
<a class="sourceLine" id="cb347-20" title="20">      <span class="dt">g=</span>stat_data<span class="op">$</span>condition,</a>
<a class="sourceLine" id="cb347-21" title="21">      <span class="dt">p.adjust.method=</span><span class="st">&quot;bonferroni&quot;</span>,</a>
<a class="sourceLine" id="cb347-22" title="22">      <span class="dt">exact=</span><span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb347-23" title="23">    )</a>
<a class="sourceLine" id="cb347-24" title="24">  )</a>
<a class="sourceLine" id="cb347-25" title="25">}</a></code></pre></div>
<pre><code>## [1] &quot;PENALTY: 0&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  frac_hitchhiking_linked_trait_change by condition
## Kruskal-Wallis chi-squared = 211.29, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$frac_hitchhiking_linked_trait_change and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC &lt;2e-16      -      
## STATIC  &lt;2e-16      0.031  
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.03&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  frac_hitchhiking_linked_trait_change by condition
## Kruskal-Wallis chi-squared = 186.88, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$frac_hitchhiking_linked_trait_change and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 2.9e-16     -      
## STATIC  &lt; 2e-16     -      
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.1&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  frac_hitchhiking_linked_trait_change by condition
## Kruskal-Wallis chi-squared = 113.72, df = 2, p-value &lt; 2.2e-16
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$frac_hitchhiking_linked_trait_change and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 3.3e-08     -      
## STATIC  &lt; 2e-16     -      
## 
## P value adjustment method: bonferroni 
## [1] &quot;PENALTY: 0.3&quot;
## 
##  Kruskal-Wallis rank sum test
## 
## data:  frac_hitchhiking_linked_trait_change by condition
## Kruskal-Wallis chi-squared = 34.791, df = 2, p-value = 2.788e-08
## 
## 
##  Pairwise comparisons using Wilcoxon rank sum test with continuity correction 
## 
## data:  stat_data$frac_hitchhiking_linked_trait_change and stat_data$condition 
## 
##         NON-PLASTIC PLASTIC
## PLASTIC 0.26        -      
## STATIC  2.4e-08     0.18   
## 
## P value adjustment method: bonferroni</code></pre>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb349-1" title="1">denom &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;NON-PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases)</a>
<a class="sourceLine" id="cb349-2" title="2">num &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;NON-PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_with_primary_trait_change)</a>
<a class="sourceLine" id="cb349-3" title="3"><span class="kw">paste0</span>(<span class="st">&quot;NON-PLASTIC (0.1 penalty): &quot;</span>, num<span class="op">/</span>denom, <span class="st">&quot;(&quot;</span>, num, <span class="st">&quot;/&quot;</span>, denom, <span class="st">&quot;)&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;NON-PLASTIC (0.1 penalty): 0.498956158663883(956/1916)&quot;</code></pre>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb351-1" title="1">denom &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases)</a>
<a class="sourceLine" id="cb351-2" title="2">num &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_with_primary_trait_change)</a>
<a class="sourceLine" id="cb351-3" title="3"><span class="kw">paste0</span>(<span class="st">&quot;PLASTIC (0.1 penalty): &quot;</span>, num<span class="op">/</span>denom, <span class="st">&quot; (&quot;</span>, num, <span class="st">&quot;/&quot;</span>, denom, <span class="st">&quot;)&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;PLASTIC (0.1 penalty): 0 (0/18)&quot;</code></pre>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb353-1" title="1">denom &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;STATIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases)</a>
<a class="sourceLine" id="cb353-2" title="2">num &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;STATIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_with_primary_trait_change)</a>
<a class="sourceLine" id="cb353-3" title="3"><span class="kw">paste0</span>(<span class="st">&quot;STATIC (0.1 penalty): &quot;</span>, num<span class="op">/</span>denom, <span class="st">&quot; (&quot;</span>, num, <span class="st">&quot;/&quot;</span>, denom, <span class="st">&quot;)&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;STATIC (0.1 penalty): 0 (0/58)&quot;</code></pre>
<p>Focal figure for the manuscript:</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb355-1" title="1"><span class="co"># Compute manual labels for geom_signif</span></a>
<a class="sourceLine" id="cb355-2" title="2">stat.test &lt;-<span class="kw">filter</span>(focal_summary_data,dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">&gt;</span><span class="dv">0</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-3" title="3"><span class="st">  </span><span class="kw">wilcox_test</span>(frac_hitchhiking_linked_trait_change <span class="op">~</span><span class="st"> </span>condition, <span class="dt">comparisons=</span><span class="kw">list</span>(<span class="kw">c</span>(<span class="st">&quot;PLASTIC&quot;</span>, <span class="st">&quot;NON-PLASTIC&quot;</span>), <span class="kw">c</span>(<span class="st">&quot;STATIC&quot;</span>, <span class="st">&quot;NON-PLASTIC&quot;</span>))) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-4" title="4"><span class="st">  </span><span class="kw">adjust_pvalue</span>(<span class="dt">method =</span> <span class="st">&quot;bonferroni&quot;</span>) <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-5" title="5"><span class="st">  </span><span class="kw">add_significance</span>() <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb355-6" title="6"><span class="st">  </span><span class="kw">add_xy_position</span>(<span class="dt">x=</span><span class="st">&quot;condition&quot;</span>)</a>
<a class="sourceLine" id="cb355-7" title="7"><span class="co"># Tweak y.position manually to account for scaled axis (edge case that triggers bad behavior in geom_signif)</span></a>
<a class="sourceLine" id="cb355-8" title="8">stat.test<span class="op">$</span>manual_position &lt;-<span class="st">  </span>stat.test<span class="op">$</span>y.position</a>
<a class="sourceLine" id="cb355-9" title="9">stat.test<span class="op">$</span>label &lt;-<span class="st"> </span><span class="kw">mapply</span>(p_label,stat.test<span class="op">$</span>p.adj)</a>
<a class="sourceLine" id="cb355-10" title="10"></a>
<a class="sourceLine" id="cb355-11" title="11">linked_trait_change_fig &lt;-<span class="st"> </span><span class="kw">ggplot</span>(</a>
<a class="sourceLine" id="cb355-12" title="12">    <span class="kw">filter</span>(focal_summary_data, dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">&gt;</span><span class="dv">0</span>),</a>
<a class="sourceLine" id="cb355-13" title="13">    <span class="kw">aes</span>(<span class="dt">x=</span>condition, <span class="dt">y=</span>frac_hitchhiking_linked_trait_change, <span class="dt">fill=</span>condition)</a>
<a class="sourceLine" id="cb355-14" title="14">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-15" title="15"><span class="st">  </span><span class="kw">geom_flat_violin</span>(</a>
<a class="sourceLine" id="cb355-16" title="16">    <span class="dt">scale=</span><span class="st">&quot;width&quot;</span>,</a>
<a class="sourceLine" id="cb355-17" title="17">    <span class="dt">position =</span> <span class="kw">position_nudge</span>(<span class="dt">x =</span> <span class="fl">.2</span>, <span class="dt">y =</span> <span class="dv">0</span>),</a>
<a class="sourceLine" id="cb355-18" title="18">    <span class="dt">alpha =</span> <span class="fl">.8</span></a>
<a class="sourceLine" id="cb355-19" title="19">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-20" title="20"><span class="st">  </span><span class="kw">geom_point</span>(</a>
<a class="sourceLine" id="cb355-21" title="21">    <span class="dt">mapping=</span><span class="kw">aes</span>(<span class="dt">color=</span>condition),</a>
<a class="sourceLine" id="cb355-22" title="22">    <span class="dt">position =</span> <span class="kw">position_jitter</span>(<span class="dt">width =</span> <span class="fl">.15</span>),</a>
<a class="sourceLine" id="cb355-23" title="23">    <span class="dt">size =</span> <span class="fl">.5</span>,</a>
<a class="sourceLine" id="cb355-24" title="24">    <span class="dt">alpha =</span> <span class="fl">0.8</span></a>
<a class="sourceLine" id="cb355-25" title="25">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-26" title="26"><span class="st">  </span><span class="kw">geom_boxplot</span>(</a>
<a class="sourceLine" id="cb355-27" title="27">    <span class="dt">width =</span> <span class="fl">.1</span>,</a>
<a class="sourceLine" id="cb355-28" title="28">    <span class="dt">outlier.shape =</span> <span class="ot">NA</span>,</a>
<a class="sourceLine" id="cb355-29" title="29">    <span class="dt">alpha =</span> <span class="fl">0.5</span></a>
<a class="sourceLine" id="cb355-30" title="30">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-31" title="31"><span class="st">  </span><span class="kw">scale_x_discrete</span>(</a>
<a class="sourceLine" id="cb355-32" title="32">    <span class="dt">name=</span><span class="st">&quot;Condition&quot;</span>,</a>
<a class="sourceLine" id="cb355-33" title="33">    <span class="dt">limits=</span>condition_order,</a>
<a class="sourceLine" id="cb355-34" title="34">    <span class="dt">labels=</span>condition_order</a>
<a class="sourceLine" id="cb355-35" title="35">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-36" title="36"><span class="st">  </span><span class="kw">scale_y_continuous</span>(</a>
<a class="sourceLine" id="cb355-37" title="37">    <span class="dt">name=</span><span class="st">&quot;Fraction of linked poisonous task acquisition&quot;</span>,</a>
<a class="sourceLine" id="cb355-38" title="38">    <span class="dt">limits=</span><span class="kw">c</span>(<span class="op">-</span><span class="fl">0.01</span>, <span class="fl">1.2</span>),</a>
<a class="sourceLine" id="cb355-39" title="39">    <span class="dt">breaks=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="fl">0.25</span>, <span class="fl">0.50</span>, <span class="fl">0.75</span>, <span class="fl">1.0</span>)</a>
<a class="sourceLine" id="cb355-40" title="40">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-41" title="41"><span class="st">  </span><span class="kw">scale_fill_brewer</span>(</a>
<a class="sourceLine" id="cb355-42" title="42">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb355-43" title="43">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-44" title="44"><span class="st">  </span><span class="kw">scale_color_brewer</span>(</a>
<a class="sourceLine" id="cb355-45" title="45">    <span class="dt">palette=</span>cb_palette</a>
<a class="sourceLine" id="cb355-46" title="46">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-47" title="47"><span class="st">  </span><span class="kw">labs</span>(</a>
<a class="sourceLine" id="cb355-48" title="48">    <span class="dt">subtitle=</span><span class="kw">paste0</span>(</a>
<a class="sourceLine" id="cb355-49" title="49">      <span class="st">&quot;Kruskal-Wallis, &quot;</span>,</a>
<a class="sourceLine" id="cb355-50" title="50">      <span class="kw">p_label</span>(<span class="kw">signif</span>(<span class="kw">kruskal.test</span>(<span class="dt">formula=</span>frac_hitchhiking_linked_trait_change<span class="op">~</span>condition, <span class="dt">data=</span>focal_summary_data)<span class="op">$</span>p.value,<span class="dt">digits=</span><span class="dv">4</span>))</a>
<a class="sourceLine" id="cb355-51" title="51">    )</a>
<a class="sourceLine" id="cb355-52" title="52">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-53" title="53"><span class="st">  </span>ggsignif<span class="op">::</span><span class="kw">geom_signif</span>(</a>
<a class="sourceLine" id="cb355-54" title="54">    <span class="dt">data=</span><span class="kw">filter</span>(stat.test, p.adj <span class="op">&lt;=</span><span class="st"> </span>alpha),</a>
<a class="sourceLine" id="cb355-55" title="55">    <span class="kw">aes</span>(<span class="dt">xmin=</span>group1,<span class="dt">xmax=</span>group2,<span class="dt">annotations=</span>label,<span class="dt">y_position=</span>manual_position),</a>
<a class="sourceLine" id="cb355-56" title="56">    <span class="dt">manual=</span><span class="ot">TRUE</span>,</a>
<a class="sourceLine" id="cb355-57" title="57">    <span class="dt">inherit.aes=</span><span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb355-58" title="58">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb355-59" title="59"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb355-60" title="60">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb355-61" title="61">  )</a></code></pre></div>
<pre><code>## Warning: Ignoring unknown aesthetics: xmin, xmax, annotations, y_position</code></pre>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb357-1" title="1">linked_trait_change_fig</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-88-1.png" width="672" /></p>
</div>
</div>
<div id="what-fraction-of-poison-execution-increases-occur-in-unexpressed-phenotype-as-cryptic-variation" class="section level2">
<h2><span class="header-section-number">7.7</span> What fraction of poison execution increases occur in unexpressed phenotype (as cryptic variation)?</h2>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb358-1" title="1"><span class="kw">ggplot</span>(<span class="kw">filter</span>(summary_data, dominant_lineage_num_times_hitchhike_inst_exec_increases<span class="op">&gt;</span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span>), <span class="kw">aes</span>(<span class="dt">x=</span>frac_unexpressed_hitchhiker_inc)) <span class="op">+</span></a>
<a class="sourceLine" id="cb358-2" title="2"><span class="st">  </span><span class="kw">geom_density</span>() <span class="op">+</span></a>
<a class="sourceLine" id="cb358-3" title="3"><span class="st">  </span><span class="kw">facet_grid</span>(</a>
<a class="sourceLine" id="cb358-4" title="4">    condition<span class="op">~</span>POISON_PENALTY,</a>
<a class="sourceLine" id="cb358-5" title="5">    <span class="dt">labeller=</span>label_both,</a>
<a class="sourceLine" id="cb358-6" title="6">    <span class="dt">scales=</span><span class="st">&quot;free_y&quot;</span></a>
<a class="sourceLine" id="cb358-7" title="7">  ) <span class="op">+</span></a>
<a class="sourceLine" id="cb358-8" title="8"><span class="st">  </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb358-9" title="9">    <span class="dt">legend.position=</span><span class="st">&quot;none&quot;</span></a>
<a class="sourceLine" id="cb358-10" title="10">  )</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-89-1.png" width="672" /></p>
<div class="sourceCode" id="cb359"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb359-1" title="1">denom &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases)</a>
<a class="sourceLine" id="cb359-2" title="2">num &lt;-<span class="st"> </span><span class="kw">sum</span>(<span class="kw">filter</span>(summary_data, condition<span class="op">==</span><span class="st">&quot;PLASTIC&quot;</span> <span class="op">&amp;</span><span class="st"> </span>POISON_PENALTY<span class="op">==</span><span class="fl">0.1</span>)<span class="op">$</span>dominant_lineage_num_times_hitchhike_inst_exec_increases_in_unexpressed_phenotype)</a>
<a class="sourceLine" id="cb359-3" title="3"><span class="kw">paste0</span>(<span class="st">&quot;PLASTIC: &quot;</span>, num<span class="op">/</span>denom, <span class="st">&quot; (&quot;</span>, num, <span class="st">&quot;/&quot;</span>, denom, <span class="st">&quot;)&quot;</span>)</a></code></pre></div>
<pre><code>## [1] &quot;PLASTIC: 0.0555555555555556 (1/18)&quot;</code></pre>
</div>
<div id="manuscript-figures-2" class="section level2">
<h2><span class="header-section-number">7.8</span> Manuscript figures</h2>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb361-1" title="1">grid &lt;-<span class="st"> </span><span class="kw">plot_grid</span>(</a>
<a class="sourceLine" id="cb361-2" title="2">  poison_increases_fig <span class="op">+</span></a>
<a class="sourceLine" id="cb361-3" title="3"><span class="st">    </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb361-4" title="4">      <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>()</a>
<a class="sourceLine" id="cb361-5" title="5">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb361-6" title="6"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Poisonous task acquisition count&quot;</span>),</a>
<a class="sourceLine" id="cb361-7" title="7">  poison_increases_per_gen_fig <span class="op">+</span></a>
<a class="sourceLine" id="cb361-8" title="8"><span class="st">    </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb361-9" title="9">      <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>()</a>
<a class="sourceLine" id="cb361-10" title="10">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb361-11" title="11"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Poisonous task acquisition frequency&quot;</span>),</a>
<a class="sourceLine" id="cb361-12" title="12">  linked_trait_change_fig <span class="op">+</span></a>
<a class="sourceLine" id="cb361-13" title="13"><span class="st">    </span><span class="kw">theme</span>(</a>
<a class="sourceLine" id="cb361-14" title="14">      <span class="dt">axis.title.x=</span><span class="kw">element_blank</span>()</a>
<a class="sourceLine" id="cb361-15" title="15">    ) <span class="op">+</span></a>
<a class="sourceLine" id="cb361-16" title="16"><span class="st">    </span><span class="kw">ggtitle</span>(<span class="st">&quot;Linked poisonous task acquisition&quot;</span>),</a>
<a class="sourceLine" id="cb361-17" title="17">  <span class="dt">nrow=</span><span class="dv">1</span>,</a>
<a class="sourceLine" id="cb361-18" title="18">  <span class="dt">align=</span><span class="st">&quot;v&quot;</span>,</a>
<a class="sourceLine" id="cb361-19" title="19">  <span class="dt">labels=</span><span class="st">&quot;auto&quot;</span></a>
<a class="sourceLine" id="cb361-20" title="20">)</a>
<a class="sourceLine" id="cb361-21" title="21"><span class="kw">save_plot</span>(</a>
<a class="sourceLine" id="cb361-22" title="22">   <span class="kw">paste0</span>(working_directory, <span class="st">&quot;plots/&quot;</span>, <span class="st">&quot;poison-accumulation-panel.pdf&quot;</span>),</a>
<a class="sourceLine" id="cb361-23" title="23">   grid,</a>
<a class="sourceLine" id="cb361-24" title="24">   <span class="dt">base_height=</span><span class="dv">6</span>,</a>
<a class="sourceLine" id="cb361-25" title="25">   <span class="dt">base_asp=</span><span class="dv">3</span><span class="op">/</span><span class="dv">1</span></a>
<a class="sourceLine" id="cb361-26" title="26">)</a>
<a class="sourceLine" id="cb361-27" title="27">grid</a></code></pre></div>
<p><img src="supplemental-material_files/figure-html/unnamed-chunk-91-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="evolution-and-maintenance-of-novel-traits.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="regulation-in-avida.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/amlalejini/evolutionary-consequences-of-plasticity/tree/master/experiments/2021-02-05-hitchhiking/analysis/2021-02-05-hitchhiking.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["supplemental-material.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
